plugins/plugin-watermark/src/client/composables/watermark.ts

```diff
@@ -0,0 +1,111 @@
+import { isFunction, isLinkHttp, wait } from '@vuepress/helper/client'
+import { computed, isRef, nextTick, ref, toValue, watch } from 'vue'
+import type { MaybeRefOrGetter } from 'vue'
+import { usePageFrontmatter, useRoutePath, withBase } from 'vuepress/client'
+// https://github.com/zhensherlock/watermark-js-plus/issues/579
+import { Watermark } from 'watermark-js-plus/dist/index.esm.js'
+import type {
+  WatermarkOptions,
+  WatermarkPluginOptions,
+} from '../../shared/index.js'
+
+declare const __VUEPRESS_PLUGIN_WATERMARK_OPTIONS__: Omit<
+  WatermarkPluginOptions,
+  'pageFilter'
+>
+
+const {
+  global = true,
+  delay = 500,
+  watermarkOptions: options,
+} = __VUEPRESS_PLUGIN_WATERMARK_OPTIONS__
+
+const userOptions = ref<WatermarkOptions>({})
+
+export function defineWatermarkConfig(
+  userConfig: MaybeRefOrGetter<WatermarkOptions>,
```

Comment:
A maybe ref could be a good approach at a glance, maybe you can open prs to support this in other plugins.

Reply:
I will add it to my to-do list.